packages/grid/x-data-grid-pro/src/tests/exportExcel.DataGridPro.test.tsx

```diff
@@ -0,0 +1,221 @@
+    return (
+      <div style={{ width: 300, height: 300 }}>
+        <DataGridPro {...baselineProps} apiRef={apiRef} {...props} />
+      </div>
+    );
+  };
+
+  describe('experimental feature', () => {
+    it('should not work if not enabled', () => {
+      render(<TestCaseExcelExport />);
+
+      expect(() => {
+        apiRef.current.getDataAsExcel();
+      }).toErrorDev(
+        [
+          'MUI: Excel export is experimental and must be activated by setting',
+          '<DataGridPro experimentalFeatures={{ excelExport: true }} />',
+        ].join('\n'),
+      );
+    });
+
+    it('should not display export option if not enabled', () => {
+      render(<TestCaseExcelExport components={{ Toolbar: GridToolbar }} />);
+
+      fireEvent.click(screen.queryByRole('button', { name: 'Export' }));
+      expect(screen.queryByRole('menuitem', { name: 'Print' })).not.to.equal(null);
+      expect(screen.queryByRole('menuitem', { name: 'Download as Excel' })).to.equal(null);
+    });
+
+    it('should generate a file if enabled', async () => {
+      render(<TestCaseExcelExport experimentalFeatures={{ excelExport: true }} />);
+      expect(apiRef.current.getDataAsExcel()).not.to.equal(null);
+    });
+
+    it('should display export option if not enabled', () => {
+      render(
+        <TestCaseExcelExport
+          experimentalFeatures={{ excelExport: true }}
+          components={{ Toolbar: GridToolbar }}
+        />,
+      );
+
+      fireEvent.click(screen.queryByRole('button', { name: 'Export' }));
+      expect(screen.queryByRole('menu')).not.to.equal(null);
+      expect(screen.queryByRole('menuitem', { name: 'Download as Excel' })).not.to.equal(null);
+    });
+  });
+
+  describe('generated file', () => {
+    it(`should export column with correct type`, async () => {
+      const Test = () => {
+        apiRef = useGridApiRef();
+        return (
+          <div style={{ width: 300, height: 300 }}>
+            <DataGridPro
+              columns={[
+                { field: 'str' },
+                { field: 'nb', type: 'number' },
+                { field: 'day', type: 'date' },
+                { field: 'hour', type: 'dateTime' },
+                { field: 'bool', type: 'boolean' },
+                { field: 'option', type: 'singleSelect', valueOptions: ['Yes', 'No'] },
+              ]}
+              rows={[
+                {
+                  id: 1,
+                  str: 'test',
+                  nb: 5,
+                  day: new Date('01-01-2022'),
+                  hour: new Date('01-01-2022'),
+                  bool: false,
+                  option: 'Yes',
+                },
+              ]}
+              apiRef={apiRef}
+              experimentalFeatures={{ excelExport: true }}
+            />
+          </div>
+        );
+      };
+      render(<Test />);
+
+      const workbook = (await apiRef.current.getDataAsExcel()) as Excel.Workbook;
+      const worksheet = workbook.worksheets[0];
```

Comment:
```suggestion
      const workbook = await apiRef.current.getDataAsExcel();
      const worksheet = workbook!.worksheets[0];
```

Same for "should not export actions columns".